receiver/prometheusreceiver/metrics_receiver_honor_timestamp_test.go

```diff
@@ -0,0 +1,550 @@
+http_request_duration_seconds_bucket{le="1"} 1900 %v
+http_request_duration_seconds_bucket{le="+Inf"} 2400 %v
+http_request_duration_seconds_sum 4950 %v
+http_request_duration_seconds_count 2400 %v
+
+# HELP rpc_duration_seconds A summary of the RPC duration in seconds.
+# TYPE rpc_duration_seconds summary
+rpc_duration_seconds{quantile="0.01"} 1 %v
+rpc_duration_seconds{quantile="0.9"} 6 %v
+rpc_duration_seconds{quantile="0.99"} 8 %v
+rpc_duration_seconds_sum 4980 %v
+rpc_duration_seconds_count 900 %v
+`
+
+// honorTimestampsPage3 has higher value than previous scrape.
+// So, after seeing a failed scrape, start_timestamp should not be reset for honorTimestampsPage3
+var honorTimestampsPage3 = `
+# HELP go_threads Number of OS threads created
+# TYPE go_thread gauge
+go_threads 19 %v
+
+# HELP http_requests_total The total number of HTTP requests.
+# TYPE http_requests_total counter
+http_requests_total{method="post",code="200"} 100 %v
+http_requests_total{method="post",code="400"} 5 %v
+
+# HELP http_request_duration_seconds A histogram of the request duration.
+# TYPE http_request_duration_seconds histogram
+http_request_duration_seconds_bucket{le="0.05"} 1000 %v
+http_request_duration_seconds_bucket{le="0.5"} 1500 %v
+http_request_duration_seconds_bucket{le="1"} 2000 %v
+http_request_duration_seconds_bucket{le="+Inf"} 2500 %v
+http_request_duration_seconds_sum 5000 %v
+http_request_duration_seconds_count 2500 %v
+
+# HELP rpc_duration_seconds A summary of the RPC duration in seconds.
+# TYPE rpc_duration_seconds summary
+rpc_duration_seconds{quantile="0.01"} 1 %v
+rpc_duration_seconds{quantile="0.9"} 5 %v
+rpc_duration_seconds{quantile="0.99"} 8 %v
+rpc_duration_seconds_sum 5000 %v
+rpc_duration_seconds_count 1000 %v
+`
+
+// TestHonorTimeStampsWithTrue validates honor_timestamp configuration where all metricFamilies (and each datapoint) in the testdata has explicit timestamps.
+// TestHonorTimeStampsWithTrue does not check for scenario where timestamp is not provided to all metric families in a scrape- as those situations are rare though valid.
+
+// TestHonorTimeStampsWithTrue has testdata such that start_timestamps should not be ahead of point_timestamps.
+// For valid data- Start_Timestamp should be prior to point time stamp.
+// TestHonorTimeStampsWithTrue validates:
+// For initial scrape, which does not have prior scrape and hence prior explicit timestamp. Start_timestamp is same as point timestamp, read from the explicit timestamp given to the datapoint in the testdata.
+// Start_timestamp should be the explicit timestamp of first-time you see a particular metric
+// Start_timestamp should get reset if the current scrape has lower value than the previous scrape
+
+func TestHonorTimeStampsWithTrue(t *testing.T) {
+	setMetricsTimestamp()
+	targets := []*testData{
+		{
+			name: "target1",
+			pages: []mockPrometheusResponse{
+				{code: 200, data: honorTimestampsPage1},
+				{code: 500, data: ""},
```

Comment:
If the missing scrape results here are causing problems with the start timestamp validation, get rid of them.  Test one thing at a time.  A separate test can be constructed to verify that the behavior of the start timestamp resolution logic is correct in the presence of stale scrapes.

Reply:
@Aneurysm9 scrapeLoop(scrape.go) bypasses staleness logic for explicit timestamps, hence stale scrapes in not affecting the `verifyHonorTimeStampsTrue`. Thus in this case summary and histogram `start_timestamp` have no issues that are present in `TestCoreMetricsEndToEnd`.
However, **Counter** metrics is not following the below criteria: 
_Start_timestamp should be the explicit timestamp of first-time you see a particular metric._

Currently, counter metrics `start_timestamp` is always equal to Gauge metric explicit timestamp in every scrape.
And I think this particular thing should be adjusted in `metricAdjuster`.

Reply:
@Aneurysm9 Thanks. removed missing scrapes.